doc/contributing.rst

```diff
@@ -158,5 +158,81 @@ Happy testing!
 Connecting new machine learning libraries
 =========================================
 
-Coming soon - please stay tuned!
-
+Hosting the library
+~~~~~~~~~~~~~~~~~~~
+
+Each extension created should be a stand-alone repository, compatible with the
+`OpenML repository <https://github.com/openml/openml-python>`_.
+The extension repository should work off-the-shelf with openml-python installed.
+
+Create a `public Github repo <https://help.github.com/en/articles/create-a-repo>`_ with
+the following directory structure:
+
+::
+
+| [repo name]
+|   |-- extensions
+|   |    |-- [extension name]
+|   |    |    |-- __init__.py
+|   |    |    |-- extension.py
+|   |    |    |-- config.py (optionally)
+|   |    |-- sklearn
+|   |    |    |-- __init__.py
+|   |    |    |-- extension.py
+|   |    |-- keras
+|   |    |    |-- __init__.py
+|   |    |    |-- extension.py
+
+
+
+Content of the Library
+~~~~~~~~~~~~~~~~~~~~~~
+
+To get an understanding of what OpenML does and why support of the latest
+machine learning libraries can be beneficial for the community, please refer
+`here <https://openml.github.io/OpenML/>`_.
+
+Various `examples <https://openml.github.io/openml-python/develop/examples/index.html>`_
+can show how OpenML works in
+practice.
+Moreover `this example <https://openml.github.io/openml-python/develop/examples/
+flows_and_runs_tutorial.html>`_
+shows how scikit-learn currently works with OpenMl as an extension. The
+`sklearn extension <https://github.com/openml/openml-python/tree/develop/openml/extensions>`_
+packaged with the `openml-python <https://github.com/openml/openml-python/issues/730>`_
+repository can be used as a template/benchmark to build the extension.
+
+
+API
++++
+* The extension scripts should be importing `openml` and be able to execute any
+  function from the OpenML `API <https://openml.github.io/openml-python/develop/api.html#>`_.
+* The extension class defined here should inherit the `openml extension
+  class <https://openml.github.io/openml-python/develop/generated/openml.extensions.
+  Extension.html#openml.extensions.Extension>`_.
+* The class should have the functions from `class Extension` overloaded as required.
+  The `Sklearn Extension API <https://openml.github.io/openml-python/develop/generated/
+  openml.extensions.sklearn.SklearnExtension.html#openml.extensions.sklearn.SklearnExtension>`_
+  is a good checklist to follow.
```

Comment:
checklist -> example ?

Reply:
I wanted to say that when defining the API for the new extension, one could use the existing [sklearn API](https://openml.github.io/openml-python/develop/generated/openml.extensions.sklearn.SklearnExtension.html#openml.extensions.sklearn.SklearnExtension) to use as a checklist for which functions need to be overloaded from the base class Extension.